### PR TITLE
Update metrics examples for Envoy v1.10

### PR DESCRIPTION
**Pull request by rata**
Since the recent upgrade to envoy 1.10 there is no need to use the
statsd sink, as envoy exports those metrics by default to prometheus
now.

This was added on envoy PR:
https://github.com/envoyproxy/envoy/pull/5601 that is included in 1.10.

This commits just removes the statsd references and updates the doc.

With these changes, the statsd-enable contour flag can probably be
removed on a future patch, will create an issue for that.

Fixes: #1035
Signed-off-by: Rodrigo Campos <rodrigo@kinvolk.io>

### CODE REVIEW

**Comment by rata** [comment]
I will try to test the host-split setup on a test cluster (have test others) soon. If someone wants to help testing it, please feel free to do it :)

**Comment by stevesloka** [comment]
Hey @rata thanks for this! I want to double check the histograms quickly before merging to validate they indeed work the way they are supposed to in v1.10.

File: deployment/ds-hostnet-split/03-envoy.yaml [hunk 1]
```diff
@@ -81,7 +67,6 @@ spec:
         - $(CONTOUR_SERVICE_HOST)
         - --xds-port
         - $(CONTOUR_SERVICE_PORT)
-        - --statsd-enabled
```
Comment (davecheney) [inline deployment/ds-hostnet-split/03-envoy.yaml]: @rata please raise an issue to remove the `--statsd-enabled` flag from `cmd/contour` if its not longer needed.

Comment (davecheney) [reply deployment/ds-hostnet-split/03-envoy.yaml]: Oh hey, you already did, #1086. Thank you!

Comment (rata) [reply deployment/ds-hostnet-split/03-envoy.yaml]: Thank you :)

**Comment by rata** [comment]
@stevesloka please, go ahead. I've tried it and I'm using contour with envoy 1.10, and I see all the histogram metrics in a test and production environment for a few weeks already, if it helps :)

**Comment by davecheney** [comment]
summoning @youngnick for visibility.

**Comment by rata** [comment]
@stevesloka thank you! I just rebased :)

**Pull request merged**